internal/pod_client.go

```diff
@@ -32,24 +32,47 @@ import (
 	"net"
 	"net/http"
 	"os"
+	"reflect"
 	"strings"
 	"time"
 
 	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
 	"github.com/FoundationDB/fdb-kubernetes-operator/pkg/podclient"
+	monitorapi "github.com/apple/foundationdb/fdbkubernetesmonitor/api"
 	"github.com/go-logr/logr"
 	"github.com/hashicorp/go-retryablehttp"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
 )
 
+// FDBImageType describes a type of image a pod or cluster is using.
+type FDBImageType string
```

Comment:
I think it's a good time to actually split up this file to have a dedicated file for the mock, sidecar and the annotation client.

Reply:
I don't think the file is too large now that we've moved some other things out, and hopefully it will get smaller in the future once we're able to deprecate and then remove the sidecar-based client. There's also some common functions like `GetImageType` that aren't necessarily specific to one type of client, but feel right in a shared file for multiple types of clients.

Reply:
That's fair, I guess he diff view made it look bigger.